pytorch_lightning/trainer/supporters.py

```diff
@@ -120,69 +128,125 @@ def mean(self):
 
 
 class PredictionCollection(object):
-    def __init__(self, global_rank: int, world_size: int):
-        self.global_rank = global_rank
-        self.world_size = world_size
-        self.predictions = {}
-        self.num_predictions = 0
-
-    def _add_prediction(self, name, values, filename):
-        if filename not in self.predictions:
-            self.predictions[filename] = {name: values}
-        elif name not in self.predictions[filename]:
-            self.predictions[filename][name] = values
-        elif isinstance(values, Tensor):
-            self.predictions[filename][name] = torch.cat(
-                (self.predictions[filename][name], values)
-            )
-        elif isinstance(values, list):
-            self.predictions[filename][name].extend(values)
+    def __init__(self, trainer):
+        self.trainer = trainer
+        self.global_rank = self.trainer.global_rank
+        self.world_size = self.trainer.world_size
+        self._predictions = {stage: {} for stage in LoggerStages}
 
-    def add(self, predictions):
+    @property
+    def current_stage(self):
+        return self.trainer.logger_connector._current_stage
+
+    @property
+    def predictions(self):
+        return self._predictions[self.current_stage]    # type: ignore
+
+    @property
+    def should_all_gather(self):
+        return torch.distributed.is_available() or not torch.distributed.is_initialized() and self.world_size > 1
+
+    @staticmethod
+    def convert_to_numpy(value):
```

Comment:
what do we need this for? Something like that should be left to the user.

Reply:
Removed !